### DISCUSSION TITLE
v0.3.0

### DISCUSSION THREAD

**Discussion by assafelovic** [assafelovic/gpt-researcher#666] [Announcements]
We're excited to introduce our latest release that includes a completely new UX/UI frontend built with NextJS! This is a huge milestone for the GPT Researcher community, now enabling not only an in depth high quality research agent, but also an optimal research user experience. In addition the release includes some additional performance and stability improvements. Thanks again the amazing community! Check out the new frontend in action:

https://github.com/user-attachments/assets/79fad973-6fbc-4149-8cc1-7e33f0473b36

## What's Changed
* Add tone selection feature to research report by @0x11c11e in https://github.com/assafelovic/gpt-researcher/pull/654
* Fixed the issue with the generate_report_prompt function to accept the tone parameter by @0x11c11e in https://github.com/assafelovic/gpt-researcher/pull/659
* Move blocking operations to threads by @tv2-vonstring in https://github.com/assafelovic/gpt-researcher/pull/660
* react frontend 2.0 by @ElishaKay in https://github.com/assafelovic/gpt-researcher/pull/523

## New Contributors
* @tv2-vonstring made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/660

**Full Changelog**: https://github.com/assafelovic/gpt-researcher/compare/v0.2.8...v0.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/assafelovic/gpt-researcher/releases/tag/v0.3.0'>v0.3.0</a>.</em>

**Comment by crti** [comment]
Since ca. 2 weeks I (Phyton n00b) try to install GPT Researcher on a Linux VM. I do not have a subscription to OpenAI or any other LLM service and my main PC is rather slow to use local LLM (no dedicated GPU). 
I followed the Getting Started Installation instructions on Github and was able to setup TAVILY_API_KEY. 
I then tried freely available API keys from OPENAI, GEMINI, GROQ and HUGGINGFACE but GPT Researcher was newer able to complete the task, I think it had requests per time limits errors - have not logged it exactly in the previous GPT Researcher Version. 
I would not care if the research is done "slowly" and letting it run for some time. 
I tried to install this latest v0.3.0 Version by doing a fresh install but the new frontend with Settings seems to not show up. I was not able to find a Version info in the files - maybe you could include that info in the frontend? 
For now I'm again stuck and need to rely on storm @ stanford and other similar tools, but GPT Researcher seems to be more capable. 
Suggestions how to install and setup v0.3.0 for my case and future "one click installing" with free online server/services setup instructions would be appreciated. 

Log latest try
`git clone https://github.com/assafelovic/gpt-researcher.git
Cloning into 'gpt-researcher'...
remote: Enumerating objects: 10594, done.
remote: Counting objects: 100% (778/778), done.
remote: Compressing objects: 100% (332/332), done.
remote: Total 10594 (delta 469), reused 556 (delta 445), pack-reused 9816
Receiving objects: 100% (10594/10594), 12.42 MiB | 27.70 MiB/s, done.
Resolving deltas: 100% (7113/7113), done.
`

**Reply by 0x11c11e** [reply]
Have you considered using Docker? The process is very straightforward with Docker.

**Reply by crti** [reply]
Thx for the suggestion. I have 0 experience with docker so I need to lockup if Docker in a Linux VM (not huge resources) is working in an acceptable way and may try it (with free API keys). 
I went ahead with https://docs.docker.com/engine/install/
EDIT1: found out systemd is needed (was not the standard bootup option on my Debian based distro); then had to fiddle with activating docker to then stat installing with "sudo docker-compose up". This started the package install as far as I understand it, took gpt-researcher 477s (15/15) + gptr-nextjs 161s (10/10) build time 
Then GPT Researcher at http://localhost:8000/ was available. However I still do not see the v0.3.0 Version Settings button (react frontend like in the video above?)
![GTRes20240720](https://github.com/user-attachments/assets/8646a04d-4df2-4b22-b94f-60b7644391bd)
